src/NServiceBus.Core/Pipeline/Incoming/ITransportReceiveContext.cs

```diff
@@ -11,5 +12,11 @@ public interface ITransportReceiveContext : IBehaviorContext
         /// The physical message being processed.
         /// </summary>
         IncomingMessage Message { get; }
+    
+        /// <summary>
+        /// Allows the pipeline to flag that it has been aborted and the receive operation should be rolled back. 
+        /// It also allows the transport to communicate to the pipeline to abort if possible.
+        /// </summary>
+        CancellationTokenSource CancellationTokenSource { get; }
```

Comment:
>Isn't that a different concern where we should only provide a token to the pipeline?

No, we need to pipeline to be able to cancel so we need to pass the token source?